### DISCUSSION TITLE
Make folder-based routing default for Vue-router.

### DISCUSSION THREAD

**Discussion by yogeshgalav** [vuejs/rfcs#486] [General]
Problem:
My team and colleagues have wasted a lot of time in finalizing ideal folder structure,
We also wasted our time, effort and money in writing and maintaining routes manually,

Solution:
All these time and effort would have been saved if Vue documentation had suggested,
Or Vue-router had by default strategy to save 1000's of line of code.

We have known that Nuxt provides this awesome thing but didn't want to migrate.
There are Vite plugin and Vue packages available,
but it would be helpful to beginners if some official message is given directly in documentation.
**Or if folder-based routing is default in Vue-router. **
   
These would not be a breaking change as manually written routes would still work.
But would facilitate future work.

**Comment by so1ve** [comment]
I believe it is impossible for vue-router to provide this feature. After all, vue-router is a pure runtime library which does not introduce a build step. There are already many awesome plugins to make this possible.

**Reply by yogeshgalav** [reply]
Including some package as an official package gains developers trust. 
Gives a lot more visibility to that individual package,
and sets standardization in main package.

In this case, if the package like `unplugin-vue-router` is officially included in vue ecosystem as an option.
Almost all future Vue applications will looks the same in terms of folder structure.
Currently every junior vue dev has to read lot of blogs to implement their own standardized folder structure. Then they join other company to work on other standardized folder structure. 
I hope you consider developer experience here.

**Reply by ShGKme** [reply]
> Almost all future Vue applications will looks the same in terms of folder structure.

> Currently every junior vue dev has to read lot of blogs to implement their own standardized folder structure. Then they join other company to work on other standardized folder structure.

A folder with pages is just one of many folders in the application. And if an application isn't small, it is likely divided into something much more complex, than just `component`, `composables`, `pages/views` folders.

Not mentioning non-typical apps that are not just a single Vue SPA.

One of the advantages of Vue is the flexibility of building project structure and architecture that best suits the app, including non-typical apps.

> Currently every junior vue dev has to read lot of blogs to implement their own standardized folder structure. Then they join other company to work on other standardized folder structure.

There is `create-vue` CLI (`npm create vue`) with a simple default structure.

Developers willing to have less flexibility and more specified out of the box can use meta-frameworks like Nuxt. Then they have folder structure, bundling, file-based routing and etc.

In any case, real projects are always different from project to project, from team to team. And that's fine.

> I hope you consider developer experience here.

This consideration should also consider that file-based routing doesn't suit all the projects and can be found as a very implicit solution.
